Assets/Scripts/Interactable/Socket.cs

```diff
@@ -0,0 +1,60 @@
+using StarterAssets;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class Socket : Interactable
+{
+    [SerializeField]
+    private GameObject _lanternObject;
+    [SerializeField]
+    private GameObject _lanternFollowSocket;
```

Comment:
Maybe a different name for this would be appropriate as to not confuse the lanternFollow object in the PlayerCapsule with the socket the lantern goes into to solve the puzzle. I would suggest **_lanternFollowPlayer** or something like that :)